proposals/25-ipld-in-ipfs.md

```diff
@@ -0,0 +1,148 @@
+* We’d like to have access to IPLD Selectors or other IPLD querying capabilities across the web3 stack for our own and others development efforts, because they enable a natural representation for addressing individual pieces of data.
+* There are external developers who would like to work directly with structured data in our stack, as opposed to serializing to flat files prior to import. Support for structured data in IPFS is immediately useful for them.
+* go-ipld-prime is sufficiently mature to be integrated through major portions of the web 3 stack, and remaining sharp edges are most easily ironed out through integration.
+
+#### User workflow example
+
+I am a developer.
+I have a set of structured JSON data.
+Or, perhaps I have an existing Web2 JSON API.
+I want to import that data directly into IPFS.
+
+I can run:
+
+    ipfs dag import users.json
+
+Or perhaps even:
+
+    ipfs dag import  https://api.spacexdata.com/v4/launches/latest
+
+This returns a CID for the data, encoded in a block in CBOR.
+I can fetch said block and output as JSON.
+
+Let’s say the CID for our SpaceX json is `QMABCDEF`.
+Soon, on another node, I should be able to fetch that data from the CID reference (note that selector pathing is not fully encompassed in this project):
+
+    ipfs dag get QMABCDEF/links/flickr/original/0
+    
+And per the sample response [here](https://github.com/r-spacex/SpaceX-API),
+I’d get back the text `https://live.staticflickr.com/65535/49927519643_b43c6d4c44_o.jpg`.
+
+Eventually, we might choose to recognize http links, download the results, and convert them to MerkleDAG data (either more CBOR data for json or UnixFS for HTML/Gif/PNG).
+Now the same query gets back `QMGHIJKL` -- a CID to download the image above of a space shuttle launch!
+
+#### Impact
+
+We have a number of existing users who have structured data they want to put in IPFS.
+Currently, they are expending effort to take that data and put it in flat files before importing into IPFS, with almost no query capabilities. 
+
+#### Leverage
+
+This a project whose greatest impact may be unblocking other projects. Among them:
+
+* Faster data transfer in IPFS (we can integrate bitswap and graphsync to speed up transfer, we cannot in the current stack)
+* Unlocking greater capabilities in the stack through use of selectors, including to-be-written selectors
+* IPLD based databases and graph querying as optimization mechanism
+* Standardizing IPLD across the stack including Filecoin (see meta doc above)
+* Coherent IPFS and Filecoin retrieval patterns (also needs Filecoin retrieval improvements)
+* Almost any effort to support dApps that work with non-static or non-file data
+
+#### Confidence
+
+On the scale: 1 -- Low
+
+Impact: 1 -- We have anecdotal examples of users who are frustrated with their inability to put semantic data into IPFS (todo: collect examples from Eric Myhre, Adin Schmahmann)
+
+Impact 3: Top user request: https://github.com/ipfs/go-ipfs/issues/7909
```

Comment:
We're listing 2 impact numbers.  Is that intentional?  Also, should this be up in the Impact section above?

Reply:
I think the iteration we heard most recently from @anorth is that the impact score is subjective enough that we might just want the prose without trying to assign a score. I think the 2 paragraphs were different options proposed while drafting this that both ended up in the document.